src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -1596,7 +1593,7 @@ protected boolean isProperDefinition(DetailAST ident, DetailAST ast) {
         protected AbstractFrame getIfContains(DetailAST identToFind, boolean lookForMethod) {
             AbstractFrame frame = null;
 
-            if (lookForMethod && containsMethod(identToFind)
+            if (containsMethod(identToFind)
                 || containsFieldOrVariable(identToFind)) {
```

Comment:
Going back to the original mutation, the reason why this can't be killed is because of what `containsMethod` does.

It eventually calls `isSimilarSignature`. Assuming both are methods, this checks the parameter count of the method calls to see if they are identical. When we give it a variable, there are no parameters, so the count never lines up and this method always fails.

This is another example of us blocking from doing more than necessary computations.